tree/dataframe/inc/ROOT/RDF/RAction.hxx

```diff
@@ -98,11 +98,22 @@ public:
       fHelper.InitTask(r, slot);
    }
 
+   template <typename ColType>
+   auto GetValueChecked(unsigned int slot, std::size_t readerIdx, Long64_t entry) -> ColType &
+   {
+      if (auto *val = fValues[slot][readerIdx]->template TryGet<ColType>(entry))
+         return *val;
+
+      throw std::runtime_error{"RDataFrame: Action (" + fHelper.GetActionName() +
```

Comment:
- I like the `std::out_of_range` alternative. 
- I agree we could add a statement saying something like: "use DefaultFor to provide a default value or SkipMissingEntries if you want to filter out missing values instead."
- Regarding the possibility to print more info, I don't see what we could add *easily*. Ideally, I would like to give the user info about which part of their dataset is missing their value (i.e. not just the column name but also the specific TTree missing it). But I don't know of an easy way (yet), I have an idea in mind but it would probably take some more code on top of the already existing refactoring.